rtmp-client/src/main/java/net/butterflytv/rtmp_client/RtmpClient.java

```diff
@@ -131,19 +352,23 @@ private native int nativeOpen(String url, boolean isPublishMode, long rtmpPointe
      *  @throws IOException if connection is not opened or connection to server is lost
      *
      */
-    public int read(byte[] data, int offset, int size) throws IOException {
-        return nativeRead(data, offset, size, rtmpPointer);
+    public int read(byte[] data, int offset, int size) throws IOException, IllegalStateException {
+        int ret = nativeRead(data, offset, size, rtmpPointer);
+        if (ret < RtmpIOException.RTMP_ERROR_NONE && ret != RtmpIOException.RTMP_READ_DONE) {
```

Comment:
nit - Use `!=` instead of `<`, so the reader of the code doesn't have to understand all error codes are smaller than `RTMP_ERROR_NONE` to understand the code.

Note: Probably needs removal of `RTMP_SUCCESS`.

Reply:
Same as write()

Reply:
Oops; please ignore.